internal/feather.go

```diff
@@ -112,64 +124,90 @@ func setBuilderValues(builder array.Builder, slice reflect.Value, nullMask []boo
 		}
-		builder.(*array.Int64Builder).AppendValues(convertedValues, nullMask)
+		builder.(*array.Int64Builder).AppendValues(convertedValues, valid)
 	case reflect.Int8:
-		builder.(*array.Int8Builder).AppendValues(values.([]int8), nullMask)
+		builder.(*array.Int8Builder).AppendValues(values.([]int8), valid)
 	case reflect.Int16:
-		builder.(*array.Int16Builder).AppendValues(values.([]int16), nullMask)
+		builder.(*array.Int16Builder).AppendValues(values.([]int16), valid)
 	case reflect.Int32:
-		builder.(*array.Int32Builder).AppendValues(values.([]int32), nullMask)
+		builder.(*array.Int32Builder).AppendValues(values.([]int32), valid)
 	case reflect.Int64:
-		builder.(*array.Int64Builder).AppendValues(values.([]int64), nullMask)
+		builder.(*array.Int64Builder).AppendValues(values.([]int64), valid)
 	case reflect.Uint:
 		arrayValues := values.([]uint)
 		convertedValues := []uint64{}
 		for _, value := range arrayValues {
 			convertedValues = append(convertedValues, uint64(value))
 		}
-		builder.(*array.Uint64Builder).AppendValues(convertedValues, nullMask)
+		builder.(*array.Uint64Builder).AppendValues(convertedValues, valid)
 	case reflect.Uint8:
-		builder.(*array.Uint8Builder).AppendValues(values.([]uint8), nullMask)
+		builder.(*array.Uint8Builder).AppendValues(values.([]uint8), valid)
 	case reflect.Uint16:
-		builder.(*array.Uint16Builder).AppendValues(values.([]uint16), nullMask)
+		builder.(*array.Uint16Builder).AppendValues(values.([]uint16), valid)
 	case reflect.Uint32:
-		builder.(*array.Uint32Builder).AppendValues(values.([]uint32), nullMask)
+		builder.(*array.Uint32Builder).AppendValues(values.([]uint32), valid)
 	case reflect.Uint64:
-		builder.(*array.Uint64Builder).AppendValues(values.([]uint64), nullMask)
+		builder.(*array.Uint64Builder).AppendValues(values.([]uint64), valid)
 	case reflect.Float32:
-		builder.(*array.Float32Builder).AppendValues(values.([]float32), nullMask)
+		builder.(*array.Float32Builder).AppendValues(values.([]float32), valid)
 	case reflect.Float64:
-		builder.(*array.Float64Builder).AppendValues(values.([]float64), nullMask)
+		builder.(*array.Float64Builder).AppendValues(values.([]float64), valid)
 	case reflect.String:
-		builder.(*array.LargeStringBuilder).AppendValues(values.([]string), nullMask)
+		builder.(*array.LargeStringBuilder).AppendValues(values.([]string), valid)
 	case reflect.Bool:
-		builder.(*array.BooleanBuilder).AppendValues(values.([]bool), nullMask)
+		builder.(*array.BooleanBuilder).AppendValues(values.([]bool), valid)
+	case reflect.Slice:
+		var offsets []int64
+		innerSliceType := slice.Type().Elem()
+		if innerSliceType.Kind() != reflect.Slice {
+			return errors.Errorf(
+				"expected slice of slices, instead found slice of '%s'",
+				innerSliceType.Kind(),
+			)
+		}
+		flatSlice := reflect.MakeSlice(innerSliceType, 0, slice.Len())
+		highwaterMark := 0
+		offsets = append(offsets, 0)
+		for i := 0; i < slice.Len(); i++ {
+			innerSlice := slice.Index(i)
```

Comment:
What happens if there is a null list element? Do we need to check to see if slice[i] is nil, or do nil inner slices always have length 0?